### PR TITLE
(test): useOnClickOutside test

### PR DESCRIPTION
**Pull request by jnsereko**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [x] My work includes tests or is validated by existing tests.
- [x] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

### CODE REVIEW

**Comment by jnsereko** [comment]
@brandones @ibacher 
This is failing.  It is complaining `Invalid hook call. Hooks can only be called inside of the body of a function component` 
at https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-react-utils/src/useOnClickOutside.tsx#L7

image

How can i solve this?  Do i need to  pass the `ref` as a parameter instead of declaring it using the hook(which is the cause of the error)

**Comment by ibacher** [comment]
The simplest solution is probably just to leverage the react-hooks testing library: https://react-hooks-testing-library.com/, which is made to solve this exact problem.

**Comment by nanfuka** [comment]
you could pass that hook in the component where you used that function **useOnClickOutside**. then in the function just pass that hook as a property...

File: packages/framework/esm-react-utils/src/useOnClickOutside.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  test("should call handler() when clicking outside", async () => {
+    expect(handler).not.toHaveBeenCalled();
+
+    await act(async () => {
+      render(<div ref={ref as React.MutableRefObject<HTMLDivElement>}></div>);
+      fireEvent.mouseDown(document.body);
+    });
+
+    expect(handler).toHaveBeenCalledTimes(1);
```
Comment (jnsereko) [inline packages/framework/esm-react-utils/src/useOnClickOutside.test.tsx:20]: hey @ibacher 
This line is failing.  its like `useOnClickOutside` is not called

**Pull request merged**